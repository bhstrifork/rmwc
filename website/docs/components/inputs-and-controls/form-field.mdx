---
sidebar_label: Form Field
title: ''
---

import FormField from '@rmwc/formfield/README.mdx';

<FormField />
