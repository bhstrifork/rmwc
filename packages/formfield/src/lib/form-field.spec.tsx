import { render, screen } from '@testing-library/react';
import React from 'react';
import { FormField } from './form-field';

describe('FormField', () => {
  it('renders', () => {
    const { asFragment } = render(
      <FormField>
        <input type="checkbox" id="input" />
        <label htmlFor="input">Input Label</label>
      </FormField>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with alignEnd', () => {
    const { asFragment } = render(
      <FormField alignEnd>
        <input type="checkbox" id="input" />
        <label htmlFor="input">Input Label</label>
      </FormField>
    );
    expect(screen.getByText('Input Label').parentElement).toHaveClass(
      'mdc-form-field--align-end'
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with noWrap', () => {
    const { asFragment } = render(
      <FormField noWrap>
        <input type="checkbox" id="input" />
        <label htmlFor="input">Input Label</label>
      </FormField>
    );
    expect(screen.getByText('Input Label').parentElement).toHaveClass(
      'mdc-form-field--nowrap'
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with spaceBetween', () => {
    const { asFragment } = render(
      <FormField spaceBetween>
        <input type="checkbox" id="input" />
        <label htmlFor="input">Input Label</label>
      </FormField>
    );
    expect(screen.getByText('Input Label').parentElement).toHaveClass(
      'mdc-form-field--space-between'
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
