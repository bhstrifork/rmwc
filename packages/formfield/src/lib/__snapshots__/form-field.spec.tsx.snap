// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField renders 1`] = `
<DocumentFragment>
  <div
    class="mdc-form-field"
  >
    <input
      id="input"
      type="checkbox"
    />
    <label
      for="input"
    >
      Input Label
    </label>
  </div>
</DocumentFragment>
`;

exports[`FormField renders with alignEnd 1`] = `
<DocumentFragment>
  <div
    class="mdc-form-field mdc-form-field--align-end"
  >
    <input
      id="input"
      type="checkbox"
    />
    <label
      for="input"
    >
      Input Label
    </label>
  </div>
</DocumentFragment>
`;

exports[`FormField renders with noWrap 1`] = `
<DocumentFragment>
  <div
    class="mdc-form-field mdc-form-field--nowrap"
  >
    <input
      id="input"
      type="checkbox"
    />
    <label
      for="input"
    >
      Input Label
    </label>
  </div>
</DocumentFragment>
`;

exports[`FormField renders with spaceBetween 1`] = `
<DocumentFragment>
  <div
    class="mdc-form-field mdc-form-field--space-between"
  >
    <input
      id="input"
      type="checkbox"
    />
    <label
      for="input"
    >
      Input Label
    </label>
  </div>
</DocumentFragment>
`;
